src/Application.cpp

```diff
@@ -1066,7 +1069,20 @@ bool Application::unloadGame()
 
 void Application::pauseGame(bool pause)
 {
-  RA_SetPaused(pause);
+  if (!pause)
+  {
+    _fsm.resumeGame();
+    RA_SetPaused(false);
+  }
+  else if (hardcore())
+  {
+    _fsm.pauseGame();
+    RA_SetPaused(true);
+  }
+  else
+  {
+    _fsm.pauseGameNoOvl();
+  }
```

Comment:
Pause on Reset/Pause on Trigger call the `CausePause` callback, which calls `app.pauseGame`. The previous implementation just called back into the DLL to show the overlay. This change notifies the FSM to update its state appropriately, actually allowing Pause on Reset/Pause on Trigger to behave as expected.